sdk/identity/identity/images/mermaidjs/DefaultAzureCredentialAuthFlow.md

```diff
@@ -0,0 +1,28 @@
+```mermaid
+%% STEPS TO GENERATE IMAGE
+%% =======================
+%% 1. Install mermaid CLI (see https://github.com/mermaid-js/mermaid-cli/blob/master/README.md)
+%% 2. Run command: mmdc -i DefaultAzureCredentialAuthFlow.md -o DefaultAzureCredentialAuthFlow.svg
+
+flowchart LR;
+    A(Environment):::deployed ==> B(Managed Identity):::deployed ==> C(VS Code):::developer ==> D(Azure CLI):::developer ==> E(Azure PowerShell):::developer;
+
+    subgraph CREDENTIAL TYPES;
+        direction LR;
+        Deployed(Deployed service):::deployed ==> Developer(Developer):::developer;
+
+        %% Hide links between boxes in the legend by setting width to 0. The integers after "linkStyle" represent link indices.
+        linkStyle 4 stroke-width:0px;
+    end;
+
+    %% Define styles for credential type boxes
+    classDef deployed fill:#71AD4C, stroke:#71AD4C;
+    classDef developer fill:#EB7C39, stroke:#EB7C39;
+
+    %% Add API ref links to credential type boxes
+    click A "https://docs.microsoft.com/javascript/api/@azure/identity/environmentcredential?view=azure-node-latest" _blank;
+    click B "https://docs.microsoft.com/javascript/api/@azure/identity/managedidentitycredential?view=azure-node-latest" _blank;
+    click C "https://docs.microsoft.com/javascript/api/@azure/identity/visualstudiocodecredential?view=azure-node-latest" _blank;
+    click D "https://docs.microsoft.com/javascript/api/@azure/identity/azureclicredential?view=azure-node-latest" _blank;
+    click E "https://docs.microsoft.com/javascript/api/@azure/identity/azurepowershellcredential?view=azure-node-latest" _blank;
+```
```

Comment:
Some questions:

- The arrow between “Deployed service” and “Developer” looks invisible to me.
- The margin at the top looks uneven with the bottom, is there a way to fix it? It arguably doesn’t really matter though.
- It feels to me that the contrast of the font in the boxes should be greater / blacker.

Not required, just wondering.

Reply:
>The arrow between “Deployed service” and “Developer” looks invisible to me.

This is intentional. It's my attempt to create a legend for the flowchart.

> The margin at the top looks uneven with the bottom, is there a way to fix it? It arguably doesn’t really matter though.

I wasn't able to find a way to fix this.

> It feels to me that the contrast of the font in the boxes should be greater / blacker.

I agree. See the latest commit.

Reply:
ooh ok ok! Thank you so much!